### PR TITLE
Better error messages in case of `TooLarge*` exceptions in GenBCode

### PR DESCRIPTION
**Pull request by dos65**
Just a bit more informative messages for `MethodTooLargeException` and `ClassTooLargeException` instead of unhandled error with a large stacktrace.

### CODE REVIEW

**Comment by dos65** [comment]
@michelou I think it would be better to keep 64KB  for `MethodTooLargeException` because that the actual unit of measure in this case. Also, error provides that actual size of method in bytes.

**Comment by michelou** [comment]
> @michelou I think it would be better to keep 64KB for `MethodTooLargeException` because that the actual unit of measure in this case. Also, error provides that actual size of method in bytes.

I agree, that's fine as long as both are the same.

File: compiler/src/dotty/tools/backend/jvm/GenBCode.scala [hunk 1]
```diff
@@ -558,18 +560,29 @@ class GenBCodePipeline(val int: DottyBackendInterface, val primitives: DottyPrim
...
+    catch
+      case e: MethodTooLargeException =>
+        val method = 
+          s"${e.getClassName.replaceAll("/", ".")}.${e.getMethodName}"
+        val msg = 
+          s"Generated bytecode for method '$method' is too large. Size: ${e.getCodeSize} bytes. Limit is 64KB" 
+        report.error(msg)
+      case e: ClassTooLargeException =>
+        val msg = 
+          s"Class '${e.getClassName.replaceAll("/", ".")}' is too large. Constant pool size: ${e.getConstantPoolCount}. Limit is 64K"
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/backend/jvm/GenBCode.scala:584]:
```suggestion
          s"Class '${e.getClassName.replaceAll("/", ".")}' is too large. Constant pool size: ${e.getConstantPoolCount}. Limit is 64KB"
```

**Review by nicolasstucki — COMMENTED** [review]
Otherwise LGTM

**Review by nicolasstucki — APPROVED** [review]

File: compiler/src/dotty/tools/backend/jvm/GenBCode.scala (hunk 1 above)
Comment (dos65) [reply compiler/src/dotty/tools/backend/jvm/GenBCode.scala]: Actually I did a check and in this case the unit of measure is not bytes but just an amount of entries in constant pool. 
https://github.com/scala/scala-asm/blob/8ff30b68d52ec42a63677a15b30195d7973012ef/src/main/java/scala/tools/asm/ClassWriter.java#L596-L600

**Comment by nicolasstucki** [comment]
Fixes #14213

**Pull request merged**